pkg/urbit/daemon/main.c

```diff
@@ -991,6 +993,73 @@ _cw_intr_win(c3_c* han_c)
 }
 #endif
 
+/* _cw_eval_commence(): initialize and run the hoon evaluator
+*/
+static void
+_cw_eval_commence(c3_i argc, c3_c* argv[])
+{
+  //Read from stdin until an EOF is recieved
+  c3_c* line = NULL;
+  size_t len = 0;
+  ssize_t nread;
+  c3_c* evl_c = "";
+
+  while((nread = getline(&line, &len, stdin)) != -1)
+  {
+     len = asprintf(&evl_c, "%s%s", evl_c, line);
+  }
+
+  free(line);
+
+  //printf("%s", evl_c);
+
+  //Initialize the Loom and load the Ivory Pill
+  c3_d          len_d = u3_Ivory_pill_len;
+  c3_y*         byt_y = u3_Ivory_pill;
+  u3_cue_xeno*  sil_u;
+  u3_weak       pil;
+
+  u3C.wag_w |= u3o_hashless;
+  u3m_boot_lite();
+  sil_u = u3s_cue_xeno_init_with(ur_fib27, ur_fib28);
+  if ( u3_none == (pil = u3s_cue_xeno_with(sil_u, len_d, byt_y)) ) {
+    printf("*** fail _setup 1\n");
+    exit(1);
+  }
+  u3s_cue_xeno_done(sil_u);
+  if ( c3n == u3v_boot_lite(pil) ) {
+    printf("*** fail _setup 2\n");
+    exit(1);
+  }
+
+  printf("Executing hoon:\n");
+
+  //Run the input through a virtualization (u3v_wish_n) and get the ouput
+  //u3_noun res = u3m_soft(0, u3v_wish_n, u3i_string(evl_c));
+  u3_noun gat = u3v_wish("|=(a=@t (sell (slap !>(+>.$) (rain /eval a))))");
+  u3_noun res;
+  {
+    u3_noun sam = u3i_string(evl_c);
+    u3_noun cor = u3nc(u3k(u3h(gat)), u3nc(sam, u3k(u3t(u3t(gat)))));
+    res = u3m_soft(0, u3n_kick_on, cor); // Transfer cor and sam
+  }
+
+
+
+  
+  if(0 == u3h(res)){//Succuessful execution print the output
+     u3_pier_tank(0,0,u3k(u3t(res)));
+  }else{
+     u3_pier_punt_goof("Hoon Error", u3k(res)); //print stack trace error
+  }
```

Comment:
style: `else` should be on a new line, and there should be spaces around the conditional expression for `if`.

The `"Hoon Error"` caption should also probably just be `"eval"`, it'll print something like:

```
eval: bail: %meme
<stack trace>
```

Reply:
Done